core/src/ModelArray.cpp

```diff
@@ -37,8 +37,12 @@ ModelArray::ModelArray(const ModelArray& orig)
 
 ModelArray& ModelArray::operator=(const ModelArray& orig)
 {
-    type = orig.type;
-    setData(orig.m_data);
+    if (orig.nComponents() == 1 && nComponents() != 1) {
+        component(0) = orig.data();
```

Comment:
```suggestion
        component(0) = orig.data();
        for ( int n=1; n< nComponents(); n++) {
            component(n).setZero();
```

Like this, assigning the mean component resets the higher-order components. The idea is to avoid ending up with uninitialised values when assigning.